### PR TITLE
Migrate to typescript

### PR DESCRIPTION
**Pull request by Simonpedro**

### CODE REVIEW

File: src/components/Standings/index.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      );
+      setTournament(data);
+    };
+    fetchTournamentFromDb();
+  }, [match.params.idTournament]);
+
+  return (
+    <Grid container justify={"center"}>
+      {tournament ? (
+        <>
```
Comment (leusgalvan) [inline src/components/Standings/index.tsx]: Check if these fragments are redundant

File: src/components/Team/index.tsx [hunk 2]
```diff
@@ -29,27 +30,30 @@ const styles = theme => ({
...
-  const filterteams = gs => {
-    const user = firebase.auth.currentUser;
+  const filterteams = (teams: Team[]) => {
+    const user = firebase.getCurrentUser();
+    const userId = user && user.uid;
     const filteredByUser = filterByLoggedUser
-      ? gs.filter(g => g.members.some(m => m.userId === user.uid))
-      : gs;
+      ? teams.filter(team => team.members.some(member => member.id === userId))
+      : teams;
```
Comment (leusgalvan) [inline src/components/Team/index.tsx]: Thanks for fixing.

File: src/components/Tournament/index.tsx [hunk 3]
```diff
@@ -54,13 +61,14 @@ const Tournament = () => {
...
+      firebase.saveTournament(tournament).then(() => {
+        setTournament(null);
+        setLoading(false);
+        setTournamentName("");
+        setSelectedTeams([]);
+        //TODO redirect user to tournament/tournamentID page to see scores
+      });
   };
 
   return (
```
Comment (leusgalvan) [inline src/components/Tournament/index.tsx]: I think the outmost fragment in this file is not needed.

File: src/components/App/index.tsx [hunk 4]
```diff
@@ -27,7 +27,12 @@ const App = () => {
       <Router>
         <Switch>
           <Route exact path="/" component={Home} />
-          <PrivateRoute exact path="/mainMenu" component={MainMenu} />
+          <PrivateRoute
+            exact
+            path="/mainMenu"
+            component={MainMenu}
+            onLogout={() => {}}
```
Comment (leusgalvan) [inline src/components/App/index.tsx]: We might need to check if logout is working right now.

**Comment by leusgalvan** [comment]
Left some comments. Some may not be related to this PR specifically but it's just to not forget about them!

File: src/components/Firebase/firebase.ts [hunk 5]
```diff
@@ -46,44 +50,52 @@ class Firebase {
...
+  }
+
   async fetchAllTeams() {
     const snapshot = await this.db.collection(TEAM_COLLECTION).get();
-    return snapshot.docs.map(doc => ({ ...doc.data(), id: doc.id }));
+    return snapshot.docs.map(doc => ({ ...doc.data(), id: doc.id } as Team));
   }
 
-  async saveTournament(tournament) {
+  async saveTournament(tournament: Tournament) {
```
[the remark this answers is not in the document]
Comment (Simonpedro) [reply src/components/Firebase/firebase.ts]: It depends on if the id is auto generated by the underlying storage or whether it is provided by us I suppose.

File: src/components/Firebase/firebase.ts [hunk 6]
```diff
@@ -46,44 +50,52 @@ class Firebase {
...
     const result = await this.db
       .collection(TOURNAMENT_COLLECTION)
       .add(tournament);
     return result;
   }
 
-  async fetchTournamentById(idTournament) {
+  async fetchTournamentById(
+    idTournament: string
+  ): Promise<Tournament & { id: string }> {
```
[the remark this answers is not in the document]
Comment (Simonpedro) [reply src/components/Firebase/firebase.ts]: Yeah, you are right!
cc: @ginoemiliozzi

File: src/components/Standings/index.tsx (hunk 1 above)
Comment (Simonpedro) [reply src/components/Standings/index.tsx]: Yes, they are redundant. We'd only need fragments when there are more than one component as children.

File: src/components/Team/index.tsx (hunk 2 above)
Comment (Simonpedro) [reply src/components/Team/index.tsx]: My pleasure

Comment (Simonpedro) [reply src/components/Team/index.tsx]: Gino's pleasure indeed

File: src/components/Tournament/index.tsx (hunk 3 above)
Comment (Simonpedro) [reply src/components/Tournament/index.tsx]: What fragment?

**Comment by leusgalvan** [comment]
Fixes https://github.com/leusgalvan/fixture/issues/27.

**Pull request merged**